### ISSUE TITLE
Question: Additional examples, per application keybindings?

### ISSUE THREAD

**Issue by joshgoebel** [snyball/Hawck#71]
Are there any additional docs/examples of per-application keybindings... such as say `Cmd-F` having different results depending on whether you are using Firefox or VS Code?

**Comment by cartok** [comment]
Maybe it is but not propably not yet through the DSL, i wonder though how much effort it would take to get proper identifiers for the currently focused application into the system. I got no professional experience here but I guess it can be done with wlroots but. Thinking of sway you might need to treat both cases wayland and x wayland applications, or shall it be binary name based if that is possible and maybe easier in the end?

**Comment by cartok** [comment]
Just a subset of the features here builtin to sway would be nice. Other than that or a feature for hawck, for sway, the criterias or get_tree could be used in a hawck lua script with `os.execute`?

**Comment by snyball** [comment]
@cartok Correct, you can do this with something like:

```lua
function swayIsAppFocused(name)
  local p = io.popen("put your swaymsg / jq command thing here")
  for line in p:lines() do
    -- ... check result
  end
  p:close()
  return -- ...
end

is_focused = LazyCondF.new(function (name)
  return swayIsAppFocused(name)
end)

ctrl + shift + key "c" + is_focused("chromium") => disabled
```

Refer to https://github.com/snyball/Hawck/issues/37 for the windows/super key.

**Comment by snyball** [comment]
@joshgoebel I don't think there's a widely accepted Wayland protocol for this, so you'll have to rely on specific hacks for your WM/DE of choice.

**Comment by snyball** [comment]
Actually, reopening this. If you can test and get this working @cartok I'd gladly accept a pull-request for a Sway-specific function to do this.

**Comment by joshgoebel** [comment]
> I don't think there's a widely accepted Wayland protocol for this, so you'll have to rely on specific hacks for your WM/DE of choice.

That's the feeling I'm getting and what I've heard everywhere - I just opened this to see if perhaps you knew of some magic that I do not. :-)  Shelling out to get the current window is a pretty terrible thing to need to do for every single keystroke.

**Comment by snyball** [comment]
@joshgoebel Yeah it's not ideal, although AFAIK `+` does short-circuit, so it will at least only shell out when necessary.
